scripts/snapshot.ts

```diff
@@ -0,0 +1,163 @@
+        lockDuration: await pendingWithdrawals.lockDuration(),
+        bnBntTotalSupply: await bnBNT.totalSupply()
+    };
+
+    saveCSV([stats], `./global.csv`);
+};
+
+const savePoolStats = async () => {
+    const { dai, link } = await getNamedAccounts();
+
+    const pools = {
+        ETH: NATIVE_TOKEN_ADDRESS,
+        LINK: link,
+        DAI: dai
+    };
+
+    const settings = await DeployedContracts.NetworkSettings.deployed();
+    const poolCollection = await DeployedContracts.PoolCollectionType1V1.deployed();
+    const bntPool = await DeployedContracts.BNTPool.deployed();
+    const masterVault = await DeployedContracts.MasterVault.deployed();
+    const externalProtectionVault = await DeployedContracts.ExternalProtectionVault.deployed();
+
+    const poolStats: PoolStats[] = [];
+
+    for (const [symbol, pool] of Object.entries(pools)) {
+        const token = { address: pool };
+        const poolData = await poolCollection.poolData(pool);
+        const { liquidity, averageRate } = poolData;
+        const emaGCD = gcd(averageRate.rate.n, averageRate.rate.d);
+        const poolToken = await Contracts.PoolToken.attach(poolData.poolToken);
+
+        poolStats.push({
+            symbol,
+            bntTradingLiquidity: liquidity.bntTradingLiquidity,
+            tknTradingLiquidity: liquidity.baseTokenTradingLiquidity,
+            stakedBalance: liquidity.stakedBalance,
+            depositLimit: poolData.depositLimit,
+            tradingFeePPM: poolData.tradingFeePPM,
+            depositingEnabled: poolData.depositingEnabled,
+            tradingEnabled: poolData.tradingEnabled,
+            spotRate: new Decimal(liquidity.bntTradingLiquidity.toString()).div(
+                new Decimal(liquidity.baseTokenTradingLiquidity.toString())
+            ),
+            emaRate: new Decimal(averageRate.rate.n.toString()).div(new Decimal(averageRate.rate.d.toString())),
+            emaBlockNumber: averageRate.blockNumber,
+            emaCompressedNumerator: averageRate.rate.n.div(emaGCD),
+            emaCompressedDenominator: averageRate.rate.d.div(emaGCD),
+            masterVaultTknBalance: await getBalance(token, masterVault),
+            externalProtectionVaultTknBalance: await getBalance(token, externalProtectionVault),
+            bntFundingLimit: await settings.poolFundingLimit(pool),
+            bntRemainingFunding: await bntPool.availableFunding(pool),
+            bntFundingAmount: await bntPool.currentPoolFunding(pool),
+            bnTkn: poolData.poolToken,
+            bnTknTotalSupply: await poolToken.totalSupply()
+        });
+    }
+
+    saveCSV(poolStats, `./pools.csv`);
+};
+
+const main = async () => {
+    if (!isTenderlyFork()) {
+        throw new Error('Invalid network');
+    }
+
+    await saveGlobalStats();
+    await savePoolStats();
+};
+
+main()
+    .then(() => process.exit(0))
+    .catch((error) => {
+        console.error(error);
+        process.exit(1);
+    });
```

Comment:
You can just do `main()` at the very bottom of the script as far as I know (though perhaps there is some advantage in this code which I am not aware of).

Reply:
It's a more structure way to handle and output the errors, but a plain `main` would've worked as well (with our version of node)